modules/account-lib/src/coin/hbar/tokenTransferBuilder.ts

```diff
@@ -0,0 +1,169 @@
+    const transferData = tx.txBody.cryptoTransfer;
+    if (transferData && transferData.tokenTransfers && transferData.tokenTransfers.length > 0) {
+      this.initTransfers(transferData.tokenTransfers);
+    }
+  }
+
+  /**
+   * Initialize the transfer specific data, getting the recipient account
+   * represented by the element with a positive amount on the transfer element.
+   * The negative amount represents the source account so it's ignored.
+   *
+   * @param {proto.IAccountAmount[]} transfers array of objects which contains accountID and transferred amount
+   */
+  protected initTransfers(tokenTransfers: proto.ITokenTransferList[]): void {
+    const tokenTransfer = tokenTransfers[0];
+    if (!tokenTransfer.token) {
+      throw new InvalidParameterValueError('missing token id');
+    }
+    if (!tokenTransfer.transfers) {
+      throw new InvalidParameterValueError('missing transfer data');
+    }
+    tokenTransfer.transfers.forEach((transferData) => {
+      const amount = Long.fromValue(transferData.amount!);
+      if (amount.isPositive()) {
+        this.to(stringifyAccountId(transferData.accountID!));
+        this.amount(amount.toString());
+        const tokenId = stringifyTokenId(tokenTransfer.token!);
+        if (!isValidAddress(tokenId)) {
+          throw new InvalidParameterValueError('Invalid token id');
+        }
+        this._tokenId = tokenId;
+      }
+    });
+  }
+
+  /** @inheritdoc */
+  protected signImplementation(key: BaseKey): Transaction {
+    if (this._multiSignerKeyPairs.length >= DEFAULT_M) {
+      throw new SigningError('A maximum of ' + DEFAULT_M + ' can sign the transaction.');
+    }
+    return super.signImplementation(key);
+  }
+
+  // region Transfer fields
+  /**
+   * Set the destination address where the funds will be sent,
+   * it may take the format `'<shard>.<realm>.<account>'` or `'<account>'`
+   *
+   * @param {string} address the address to transfer funds to
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  to(address: string): this {
+    if (!isValidAddress(address)) {
+      throw new InvalidParameterValueError('Invalid address');
+    }
+    this._toAddress = address;
+    return this;
+  }
+
+  /**
+   * Set the amount to be transferred
+   *
+   * @param {string} amount amount to transfer in tinyBars (there are 100,000,000 tinyBars in one Hbar)
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  amount(amount: string): this {
+    if (!isValidAmount(amount)) {
+      throw new InvalidParameterValueError('Invalid amount');
+    }
+    this._amount = amount;
+    return this;
+  }
+
+  /**
+   * Set the token id to be of the transaction
+   * it may take the format `'<shard>.<realm>.<token>'` or `'<token>'`
+   *
+   * @param {string} tokenId the token id for the token
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  tokenId(tokenId: string): this {
```

Comment:
had some consideration of using tokenName instead, but will be complicated on the `initTransfers`, so I keep tokenId instead.
But since the input of `send` in `sdk-core/wallet.ts` use tokenName, will do the mapping of tokenName to tokenId inside bitgo-microservice

Reply:
SGTM

Reply:
I think we should probably use tokenName here since all of our logic for converting between tokenName and tokenId is in BitgoJS. Mapping in bitgo-microservice just introduces more dependencies between the two repos. We can even do the mapping in this function.

Reply:
The dependancy issue is not really a problem since both map function and the tokenBuilder are in `account-lib`

But yeah tokenName input will be better